Tests/AK/TestHashFunctions.cpp

```diff
@@ -53,3 +54,53 @@ TEST_CASE(constexpr_ptr_hash)
     // "ptr_hash" test binds the result.
     static_assert(ptr_hash(FlatPtr(42)));
 }
+
+// Testing concrete hash results is not possible due to SipHash's non-determinism.
+// We instead perform some sanity checks and try to hit any asserts caused by programming errors.
+TEST_CASE(sip_hash)
+{
+    static SipHash<1, 3> StandardSipHash;
+    // default as used in the reference implementation
+    static SipHash<2, 4> UpstreamDefaultSipHash;
+    static SipHash<4, 8> SecureSipHash;
+
+    EXPECT_EQ(StandardSipHash(42), StandardSipHash(42));
+    EXPECT_EQ(UpstreamDefaultSipHash(42), UpstreamDefaultSipHash(42));
+    EXPECT_EQ(SecureSipHash(42), SecureSipHash(42));
+    EXPECT_NE(StandardSipHash(42), UpstreamDefaultSipHash(42));
+    EXPECT_NE(StandardSipHash(42), SecureSipHash(42));
+}
+
+TEST_CASE(sip_hash_bytes)
```

Comment:
I have no idea what this is testing.

Reply:
Hashing of a byte stream?

Reply:
what's all the common prefix business then?

Reply:
If we take two different but identical byte streams, they should still hash to the same thing. However, even if they have the same prefix (and the rest are 0), they don't hash to the same thing since they have differing length.

Reply:
Alright.